smoderp2d/runoff.py

```diff
@@ -231,166 +245,168 @@ def run(self):
+                        self.flow_control,
+                        self.courant
+                    )
+
+                    # stores current time step
+                    delta_t_tmp = self.delta_t
+
+                    # update time step size if necessary (based on the courant
+                    # condition)
+                    self.delta_t = self.courant.courant(self.delta_t)
+
+                    # if current time plus timestep is in next minute
+                    # of computation the dt is reduced so the next
+                    # computed time is exactly at the top of each minute
+                    oldtime = self.flow_control.total_time
+                    newtime_minute_floor = (self.flow_control.total_time + self.delta_t) // 60
+                    if newtime_minute_floor > oldtime // 60:
+                        self.delta_t = newtime_minute_floor * 60. - oldtime
+
+                    # courant conditions is satisfied (time step did
+                    # change) the iteration loop breaks
+                    if delta_t_tmp == self.delta_t:
+                        break
+
+                # Calculate actual rainfall and adds up interception todo:
+                # AP - actual is not storred in hydrographs
+                actRain = self.time_step.do_next_h(
                     self.surface,
                     self.subsurface,
-                    self.delta_t,
+                    self.rain_arr,
+                    self.cumulative,
+                    self.hydrographs,
                     self.flow_control,
-                    self.courant
+                    self.courant,
+                    potRain,
+                    self.delta_t
                 )
 
-                # stores current time step
-                delta_t_tmp = self.delta_t
-
-                # update time step size if necessary (based on the courant
-                # condition)
-                self.delta_t = self.courant.courant(self.delta_t)
-
-                # if current time plus timestep is in next minute
-                # of computation the dt is reduced so the next
-                # computed time is exactly at the top of each minute
-                oldtime = self.flow_control.total_time
-                newtime_minute_floor = (self.flow_control.total_time + self.delta_t) // 60
-                if newtime_minute_floor > oldtime // 60:
-                    self.delta_t = newtime_minute_floor * 60. - oldtime
-
-                # courant conditions is satisfied (time step did
-                # change) the iteration loop breaks
-                if delta_t_tmp == self.delta_t:
+                # if the iteration exceed the maximal amount of iteration
+                # last results are stored in hydrographs
+                # and error is raised
+                if not self.flow_control.max_iter_reached():
+                    self.hydrographs.write_hydrographs_record(
+                        0,
+                        0,
```

Comment:
@kubsta Why is `0` here now? In the explicit solution, it was `None`. Should we make two branches (`explicit -> None, implicit -> 0`) here or can we change it to `None` also for the implicit solution? As far as my tests go, the implicit solution works the same way with `0` and `None`.

Reply:
It was mistake or something from old version of SMODERP